app/cloud.go

```diff
@@ -4,14 +4,103 @@
 package app
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"time"
 
+	"github.com/mattermost/mattermost-server/v6/app/request"
 	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/i18n"
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 )
 
+func (a *App) NotifySystemAdminsToUpgrade(c *request.Context, currentUserTeamID string) *model.AppError {
+	userId := c.Session().Id
+
+	// check if already notified
+	notificationPref, err := a.Srv().Store.Preference().Get(userId, model.PreferenceCloudUserEphemeralInfo, model.CloudNotifyAdminInfo)
+	if err != nil {
+		mlog.Warn("Unable to get preference cloud_user_ephemeral_info", mlog.Err(err))
+	}
+
+	if notificationPref != nil {
+		info := &model.AdminNotificationUserInfo{}
+		err = json.Unmarshal([]byte(notificationPref.Value), info)
+		if err != nil {
+			mlog.Warn("Unable to Unmarshal", mlog.Err(err))
+		}
+
+		if !model.CanNotify(info.LastNotificationTimestamp) {
```

Comment:
Something is still ambiguous from the requirements and want to make sure we settle it before shipping it. Looks like this makes it so individual users can notify no more frequently than every 30 days. But if several users were to click notify admin within a 30 day user, the admin would get pinged, once per user. Is that acceptable? My guess from reading the ticket was we don't want that, but on closer reading I'm not sure its clear. @johndavidlugtu 

If we do want that, this looks good to me without further changes. If we don't want that, we'll need to read all the CloudNotifyAdminInfo entries in the preferences table to check for the latest timestamp (I don't think that's hard or too performance heavy).

Reply:
That’s true. In the ticket it’s not stated whether only one user can ping per 30 days. @johndavidlugtu can verify

Reply:
No, we want to edit the system bot message to update the number of people that have requested. we should not ping the admin per user request @neallred

Reply:
@neallred changed the logic. Only one user can notify within cool off period. Asking for a rereview